src/apps/chifra/internal/abis/handle_addresses.go

```diff
@@ -0,0 +1,113 @@
+package abisPkg
+
+import (
+	"context"
+	"os"
+	"sort"
+
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/abi"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/account"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/logger"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/output"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+func (opts *AbisOptions) HandleAddresses() (err error) {
+	testMode := opts.Globals.TestMode
+	result := make(abi.AbiInterfaceMap)
+	if opts.Known {
+		if err = abi.PreloadKnownAbis(opts.Globals.Chain, result, false); err != nil {
+			return
+		}
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	fetchData := func(modelChan chan types.Modeler[types.RawFunction], errorChan chan error) {
+		// result := make(abi.AbiInterfaceMap)
+		for _, addr := range opts.Addrs {
+			address := common.HexToAddress(addr)
```

Comment:
This line reveals a fundamental problem with something. It was always my intention that by the time we were "processing" stuff (that is looping) across any address arrays, they were already lower cased.

In the `options.go` code for every command, we should be doing this so that no-where in any of our code do we ever have to worry. Addresses should always be lower case, and they should be made lower case as soon as we read them in the options code.

We can discuss.

Reply:
Custom Address should solve this